tests/JobDefinitionFactoryTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests;
+
+use App\JobDefinitionFactory;
+use Keboola\DockerBundle\Docker\Component;
+use Keboola\JobQueueInternalClient\JobFactory;
+use Keboola\JobQueueInternalClient\JobFactory\Job;
+use Keboola\ObjectEncryptor\ObjectEncryptor;
+use Keboola\ObjectEncryptor\ObjectEncryptorFactory;
+use Keboola\StorageApi\Client;
+use PHPUnit\Framework\TestCase;
+
+class JobDefinitionFactoryTest extends TestCase
+{
+    public function testCreateJobDefinitionWithConfigData(): void
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+        ];
+
+        $jobDefinitions = $this->createJobDefinitions($jobData, $configData);
+
+        self::assertCount(1, $jobDefinitions);
+        $jobDefinition = $jobDefinitions[0];
+
+        self::assertSame($jobData['configId'], $jobDefinition->getConfigId());
+        self::assertSame($jobData['componentId'], $jobDefinition->getComponentId());
+        self::assertSame('bar', $jobDefinition->getConfiguration()['runtime']['foo'] ?? null);
+    }
+
+    public function testCreateJobDefinitionWithConfigDataAndBackend(): void
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+                'backend' => [
+                    'type' => 'invalid',
+                ],
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+            'backend' => [
+                'type' => 'custom',
+            ],
+        ];
+
+        $jobDefinitions = $this->createJobDefinitions($jobData, $configData);
+        self::assertSame(
+            $jobData['backend']['type'],
+            $jobDefinitions[0]->getConfiguration()['runtime']['backend']['type'] ?? null
+        );
+    }
+
+    public function testCreateJobDefinitionWithConfigId(): void
+    {
+        $configData = [
```

Comment:
název týhle proměnný je matoucí, protože configData je něco, co se zadává při spuštění jobu, tady je to prostě konfigurace

Reply:
Jo,chapu.. Je prosim te nekde zdokumetovane, co je `configuration`, co `configData`, jakou to ma strukturu a tak? Uz jsem si nad tim parkrat lamal hlavu.

Reply:
Hm, no nejlíp tak, že 
$configuration = {"id": "123", "configuration": "XXX", "name": "abc"}
a $configData = XXX

Reply:
Hmmm.. porad stejny nejasno :) Ale zkusil jsme tu hepler metody rozdelit na 2 - jedno pro testy s `configData` a jednu s `configuration`, mohlo by to ted snad byt prehlednejsi.